Source/DiabloUI/settingsmenu.cpp

```diff
@@ -0,0 +1,174 @@
+}
+
+void ItemFocused(int value)
+{
+	auto &vecItem = vecDialogItems[value];
+	optionDescription[0] = '\0';
+	if (vecItem->m_value < 0) {
+		return;
+	}
+	auto *pOption = vecOptions[vecItem->m_value];
+	auto paragraphs = WordWrapString(pOption->GetDescription(), rectDescription.size.width, GameFont12, 1);
+	strncpy(optionDescription, paragraphs.c_str(), sizeof(optionDescription));
+}
+
+void ItemSelected(int value)
+{
+	auto &vecItem = vecDialogItems[value];
+	if (vecItem->m_value < 0) {
+		auto specialMenuEntry = static_cast<SpecialMenuEntry>(vecItem->m_value);
+		switch (specialMenuEntry) {
+		case SpecialMenuEntry::PreviousMenu:
+			endMenu = true;
+			break;
+		case SpecialMenuEntry::SwitchGame:
+			gbIsHellfire = !gbIsHellfire;
+			endMenu = true;
+			recreateUI = true;
+			break;
+		case SpecialMenuEntry::ToggleSpawn:
+			gbIsSpawn = !gbIsSpawn;
+			UiSetSpawned(gbIsSpawn);
+			endMenu = true;
+			recreateUI = true;
+			break;
+		}
+		return;
+	}
+
+	auto *pOption = vecOptions[vecItem->m_value];
+	switch (pOption->GetType()) {
+	case OptionEntryType::Boolean: {
+		auto *pOptionBoolean = static_cast<OptionEntryBoolean *>(pOption);
+		pOptionBoolean->SetValue(!**pOptionBoolean);
+	} break;
+	case OptionEntryType::List: {
+		auto *pOptionList = static_cast<OptionEntryListBase *>(pOption);
+		size_t nextIndex = pOptionList->GetActiveListIndex() + 1;
+		if (nextIndex >= pOptionList->GetListSize())
+			nextIndex = 0;
+		pOptionList->SetActiveListIndex(nextIndex);
+	} break;
+	}
+	vecDialogItems[value + 1]->m_text = pOption->GetValueDescription().data();
+}
+
+void EscPressed()
+{
+	endMenu = true;
+}
+
+} // namespace
+
+void UiSettingsMenu()
+{
+	endMenu = false;
+
+	do {
+		LoadBackgroundArt("ui_art\\black.pcx");
+		LoadScrollBar();
+		UiAddBackground(&vecDialog);
+		UiAddLogo(&vecDialog);
+
+		Rectangle rectList = { { PANEL_LEFT + 50, (UI_OFFSET_Y + 204) }, { 540, 208 } };
+		rectDescription = { { PANEL_LEFT + 24, rectList.position.y + rectList.size.height + 20 }, { 590, 35 } };
+
+		optionDescription[0] = '\0';
+
+		const char *titleText = _("Settings");
+		vecDialog.push_back(std::make_unique<UiArtText>(titleText, MakeSdlRect(PANEL_LEFT, UI_OFFSET_Y + 161, PANEL_WIDTH, 35), UiFlags::FontSize30 | UiFlags::ColorUiSilver | UiFlags::AlignCenter, 8));
+		vecDialog.push_back(std::make_unique<UiScrollbar>(&ArtScrollBarBackground, &ArtScrollBarThumb, &ArtScrollBarArrow, MakeSdlRect(rectList.position.x + rectList.size.width + 5, rectList.position.y, 25, rectList.size.height)));
+		vecDialog.push_back(std::make_unique<UiArtText>(optionDescription, MakeSdlRect(rectDescription), UiFlags::FontSize12 | UiFlags::ColorUiSilverDark | UiFlags::AlignCenter, 1));
+
+		if (diabdat_mpq && hellfire_mpq)
+			vecDialogItems.push_back(std::make_unique<UiListItem>(gbIsHellfire ? _("Switch to Diablo") : _("Switch to Hellfire"), static_cast<int>(SpecialMenuEntry::SwitchGame), UiFlags::ColorUiGold));
```

Comment:
Probably change this to either "Game Mode: Hellfire", "Hellfire Expansion: On" "Hellfire: Enabled" or something along that line.